### PR TITLE
Feature/variant updates

### PR DESCRIPTION
**Pull request by botekchristophe**
Add mechanism to update Variant table when ClinVar was Updated
Add logic to execute in DEV or PROD depending on what todo. DEV execution won't publish the result and will write data to /tmp

### CODE REVIEW

File: src/main/scala/org/kidsfirstdrc/dwh/updates/UpdateVariant.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    val variant = data(Variants.TABLE_NAME).drop("clinvar_id", "clin_sig")
+    val clinvar = data("clinvar")
+    variant
+      .joinByLocus(clinvar)
+      .select(variant("*"), clinvar("name") as "clinvar_id", clinvar("clin_sig") as "clin_sig")
+  }
+
+  override def load(data: DataFrame, output: String)(implicit spark: SparkSession): DataFrame = {
+
+    val yyyyMMdd_HHmmss = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss"))
```
Comment (adipaul1981) [inline src/main/scala/org/kidsfirstdrc/dwh/updates/UpdateVariant.scala:44]: `localTimeNow`? in lieu of `yyyyMMdd_HHmmss`

File: src/main/scala/org/kidsfirstdrc/dwh/utils/MultiSourceEtlJob.scala [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+  val tableName: String
+
+  private def regexp_extractFromCreateStatement[T](regex: String, defaultValue: T)(implicit spark: SparkSession): T = {
+    Try {
+      spark.sql(s"show create table $database.$tableName")
+        .withColumn("extracted_value", regexp_extract(col("createtab_stmt"), regex, 1))
+        .where(trim(col("extracted_value")) =!= lit(""))
+        .select("extracted_value")
+        .collect().head.getAs[T](0)
+    }.toOption.getOrElse(defaultValue)
```
Comment (adipaul1981) [inline src/main/scala/org/kidsfirstdrc/dwh/utils/MultiSourceEtlJob.scala:21]: not sure if `toOption` is required.

**Review by adipaul1981 — APPROVED** [review]

File: src/main/scala/org/kidsfirstdrc/dwh/updates/Update.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  implicit val spark: SparkSession = SparkSession.builder
+    .config("hive.metastore.client.factory.class", "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory")
+    .enableHiveSupport()
+    .appName(s"Update table: $table").getOrCreate()
+
+  run(table, runEnv, rootFolder)
+
+  def run(table: String, runEnv: String, rootFolder: String)(implicit spark: SparkSession): Unit = {
+
+    val env = Try(Environment.withName(runEnv)).toOption.getOrElse(Environment.DEV)
```
[the remark this answers is not in the document]
Comment (botekchristophe) [reply src/main/scala/org/kidsfirstdrc/dwh/updates/Update.scala:20]: nope you are right

File: src/main/scala/org/kidsfirstdrc/dwh/updates/UpdateVariant.scala (hunk 1 above)
Comment (botekchristophe) [reply src/main/scala/org/kidsfirstdrc/dwh/updates/UpdateVariant.scala:44]: 👍

File: src/main/scala/org/kidsfirstdrc/dwh/utils/MultiSourceEtlJob.scala (hunk 2 above)
Comment (botekchristophe) [reply src/main/scala/org/kidsfirstdrc/dwh/utils/MultiSourceEtlJob.scala:21]: 👍

**Pull request merged**